src/plugins/autocompletion_providers/Git.ts

```diff
@@ -84,181 +141,310 @@ const notStagedFiles = unique(async(context: AutocompletionContext): Promise<Sug
+        provider: pushOptions,
+        aliases: [],
     },
     {
         name: "rebase",
         description: "Forward-port local commits to the updated upstream head.",
+        aliases: [],
     },
     {
         name: "reset",
         description: "Reset current HEAD to the specified state.",
+        aliases: [],
     },
     {
         name: "revert",
         description: "Revert some existing commits.",
+        aliases: [],
     },
     {
         name: "rm",
         description: "Remove files from the working tree and from the index.",
+        aliases: [],
     },
     {
         name: "shortlog",
         description: "Summarize git log output.",
+        aliases: [],
     },
     {
         name: "show",
         description: "Show various types of objects.",
+        aliases: [],
     },
     {
         name: "stash",
         description: "Stash the changes in a dirty working directory away.",
+        aliases: [],
     },
     {
         name: "status",
         description: "Show the working tree status.",
+        provider: statusOptions,
+        aliases: [],
     },
     {
         name: "submodule",
         description: "Initialize, update or inspect submodules.",
+        aliases: [],
     },
     {
         name: "tag",
         description: "Create, list, delete or verify a tag object signed with GPG.",
+        aliases: [],
     },
     {
         name: "worktree",
         description: "Manage multiple worktrees.",
+        aliases: [],
     },
-].map(subCommand => new Suggestion().withValue(subCommand.name).withDescription(subCommand.description).withStyle(styles.command).withSpace());
+];
+
+async function loadAllSubcommands() {
+    const text = await executeCommand("git", ["help", "-a"], process.env.HOME);
+    const matches = text.match(/  ([\-a-zA-Z0-9]+)/gm);
+
+    if (matches) {
+        matches
+            .filter((match) => match.indexOf("--") === -1)
+            .forEach(async(match) => {
+                const name = match.trim();
+                const data = find(subCommandsData, {name});
+
+                if (!data) {
+                    subCommandsData.push({
+                        name,
+                        description: "",
+                        aliases: [],
+                    });
+                }
+            });
+    }
+}
+
+async function loadAliases() {
+    const variables = await Git.configVariables(process.env.HOME);
+
+    variables
+        .filter(variable => variable.name.indexOf("alias.") === 0)
+        .forEach(variable => {
+            const name = variable.name.replace("alias.", "");
+            const data = find(subCommandsData, {name: variable.value});
+
+            if (data) {
+                data.aliases.push(name);
+            } else {
+                subCommandsData.push({
+                    name: name,
+                    description: variable.value,
+                    aliases: [],
+                });
+            }
+        });
+}
```

Comment:
Same here. Just add it to the aforementioned combine. Data mutation is one of the most widespread cause of bugs.